.github/workflows/build_snap_without_cache.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Usage documentation: build_snap_without_cache.md
+
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        description: Snap package is uploaded to this GitHub artifact name
+        default: snap-packed-without-cache
+        type: string
+      path-to-snap-project-directory:
+        description: |
+          Relative path to snap project directory from repository directory
+          
+          The "snap project directory" is the directory that contains the `snap` directory, not the `snap` directory itself.
+        default: .
+        type: string
+    outputs:
+      artifact-name:
+        description: Snap package is uploaded to this GitHub artifact name
+        value: ${{ inputs.artifact-name }}
+
+jobs:
+  build:
+    name: Build snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up environment
+        # TODO: Replace with custom image on self-hosted runner
+        run: |
+          sudo adduser "$USER" lxd
+          newgrp lxd
+          sudo lxd waitready
+          sudo lxd init --auto
+          sudo iptables -F FORWARD
+          sudo iptables -P FORWARD ACCEPT
+          sudo chmod a+wr /var/snap/lxd/common/lxd/unix.socket
```

Comment:
This is something new. JFMI, why do we need it? Security--  :-)
P.S. it is effectively "o+wr" as both user and group have rw already.